### PR TITLE
ETC 'Spiral' network upgrade

### PR DESCRIPTION
**Pull request by diega**
## PR description
This PR adds support to [ECIP-1109](https://ecips.ethereumclassic.org/ECIPs/ecip-1109).
Block number is set to https://github.com/ethereumclassic/ECIPs/pull/514, already set in core-geth in https://github.com/etclabscore/core-geth/pull/571
I also set the DNS discovery tree to what's in [core-geth#bootnodes_mordor.go](https://github.com/etclabscore/core-geth/blob/v1.12.14/params/bootnodes_mordor.go#L29)

### CODE REVIEW

**Review by shemnon — APPROVED** [review]
LGTM

**Review by jflo — APPROVED** [review]
Approved, but there is no way Spiral is in the same league as these other villains :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -96,6 +96,7 @@ https://hyperledger.jfrog.io/artifactory/besu-binaries/besu/23.7.3/besu-23.7.3.z
 - JSON output is now compact by default. This can be overridden by the new `--json-pretty-print-enabled` CLI option. [#5766](https://github.com/hyperledger/besu/pull/5766)
 - New `eth_getBlockReceipts` JSON-RPC method to retrieve all transaction receipts for a block in a single call [#5771](https://github.com/hyperledger/besu/pull/5771) 
 - Add new methods to `OperationTracer` to capture contexts enter/exit [#5756](https://github.com/hyperledger/besu/pull/5756)
+- Ethereum Classic Spiral network upgrade [#6078](https://github.com/hyperledger/besu/pull/6078)
```
Comment (macfarla) [inline CHANGELOG.md:99]: this should be in 23.10.2

**Review by macfarla — CHANGES_REQUESTED** [review]
LGTM just the changelog entry is in an old release

File: CHANGELOG.md (hunk 1 above)
Comment (diega) [reply CHANGELOG.md]: Sure, I'll move it, but do you think it would be possible to include this into `23.10.1`?

Comment (macfarla) [reply CHANGELOG.md]: sorry, that was already cut last week and burn-in just about finished. Ask in the besu-release channel re timing for the next release https://discord.com/channels/905194001349627914/943252457063075880/1164641743782805525

Comment (diega) [reply CHANGELOG.md]: @macfarla I added a new section for `23.10.2` and moved the entry there

**Pull request merged**